smk/gene_annotation.smk

```diff
@@ -188,60 +195,63 @@ rule make_cd_transl_faa_for_genome:
 # Convert GFF file into BED file ####
 
 rule make_bed_for_genome:
-    input: rules.prokka_for_genome.output.gff
+    input:
+        gff=rules.prokka_for_genome.output.gff
     output:
-        bed=                      "{wd}/DB/{post_analysis_dir}/GFF/{genome}.bed",
-        bed_hdr_mod=              "{wd}/DB/{post_analysis_dir}/GFF/{genome}.header-modif.bed",
-        bed_hdr_mod_coord_correct="{wd}/DB/{post_analysis_dir}/GFF/{genome}.header-modif.coord_correct.bed"
+        bed="{wd}/DB/{post_analysis_dir}/2-postprocessed/{genome}.bed"
     log:
         "{wd}/logs/DB/{post_analysis_dir}/{genome}.make_bed_from_gff.log"
     conda:
         config["minto_dir"]+"/envs/MIntO_base.yml" #gff2bed
     shell:
         """
-        time (\
-            gff2bed < {input} > {output.bed}
-            sed -e "s/^gnl|X|/{wildcards.genome}|/g" {output.bed} > {output.bed_hdr_mod}
-            awk -v FS='\\t' -v OFS='\\t' '{{$2+=1; print $0}}' {output.bed_hdr_mod} > {output.bed_hdr_mod_coord_correct} \
+        time (
+            # Get BED file
+            gff2bed < {input.gff} > bed.1
+
+            # Modify header
+            sed -e "s/^gnl|X|/{wildcards.genome}|/g" bed.1 > bed.2
+
+            # Fix coordinate for 0-based vs 1-based
+            awk -v FS='\\t' -v OFS='\\t' '{{$2+=1; print $0}}' bed.2 > {output.bed}
         ) >& {log}
         """
```

Comment:
If bed.1 and bed.2 are tmp filenames, as this rule is not run in a shadow folder, there is a possibility of a race condition with parallel jobs writing to the same files.

Reply:
Good catch. I keep assuming that shadow is turned on for all rules. Will fix it.